workflow-controller/controller-service/src/main/java/Converters/Converter.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package Converters;
+
+import com.github.fge.jsonschema.core.report.ProcessingMessage;
+import com.github.fge.jsonschema.core.report.ProcessingReport;
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.CommonErrorCode;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.models.Activity;
+import io.hyscale.controller.activity.ControllerActivity;
+import io.hyscale.schema.validator.SchemaProcessor;
+import io.hyscale.servicespec.commons.activity.ServiceSpecActivity;
+import io.hyscale.servicespec.commons.builder.ServiceInputType;
+import io.hyscale.servicespec.commons.exception.ServiceSpecErrorCodes;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Converter class implements CommandLine framework ITypeConverter ,
+ * performs command line input validation and assigns to the respective field.
+ */
+public abstract class Converter implements CommandLine.ITypeConverter<List<File>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Converter.class);
+
+    public abstract String getFilePattern();
+
+    public abstract String getReferenceSchema();
+
+    public abstract String getWarnMessage();
+
+    public abstract int isDataValid(File inputFile) throws Exception;
+
+    public abstract Activity getDataMistmatchActivity();
+
```

Comment:
Let the validateData itself log the error, its upto implementation what to do when validation failed. Parent class should be able to proceed to next after validation is successful.